api/nodes/serializers.py

```diff
@@ -124,8 +131,22 @@ def update(self, instance, validated_data):
         the request to be in the serializer context.
         """
         assert isinstance(instance, Node), 'instance must be a Node'
+        auth = self.get_user_auth(self.context['request'])
         for attr, value in validated_data.items():
-            setattr(instance, attr, value)
+            if attr == 'tags':
+                old_tags = set([tag._id for tag in instance.tags])
+                if value:
+                    current_tags = set(value)
+                else:
+                    current_tags = set()
+                new_tags = list(current_tags - old_tags)
```

Comment:
It doesn't seem necessary to cast these as lists. Both operations you're performing on these - membership tests and iteration - can be done on sets.

If light of this, you might consider iterative them directly, as so:

```python
for new_tag in (current_tags - old_tags):
    instance.add_tag(new_tag, auth=auth)

for deleted_tag in (old_tags - current_tags):
    instance.remove_tag(deleted_tag, auth=auth)
```

Normally I'd argue that the assignment would increase the readability, but the usage of `new_tag` and `deleted_tag` as variables during iteration (:heart:) make the purpose of the loop explicit.